app/components/new-file/new-file.component.ts

```diff
@@ -21,58 +22,46 @@ export class NewFileResult {
 })
 export class NewFileComponent implements ModalComponent, AfterViewInit {
     set modalInfo(modalInfo: ModalInfo) {
-        let plugins: string[][] = modalInfo.data;
-        let availablePlugins: any = { subLists: {} };
+        const plugins: PluginData[] = modalInfo.data;
+        // This code is extremely evil but necessary.
```

Comment:
What is it doing? It seems like its just adding a function to the plugin object rather that adding that function in core. If so it's a trivial change to put it in the right place. I may be missing something though.